src/language/syntax.rs

```diff
@@ -300,19 +300,17 @@ pub fn parse_num(state: HOPA::State) -> HOPA::Answer<Option<Box<Term>>> {
   HOPA::guard(
     Box::new(|state| {
       let (state, head) = HOPA::there_take_head(state)?;
-      Ok((state, ('0'..='9').contains(&head)))
+      Ok((state, head.is_ascii_digit()))
     }),
     Box::new(|state| {
       let (state, text) = HOPA::there_nonempty_name(state)?;
       if !text.is_empty() {
         if text.starts_with("0x") {
-          return Ok((state, Box::new(Term::U6O { numb: u60::new(u64::from_str_radix(&text[2..], 16).unwrap()) })));
+          Ok((state, Box::new(Term::U6O { numb: u60::new(u64::from_str_radix(&text[2..], 16).unwrap()) })))
```

Comment:
I'm surprised you added #[warn(clippy::manual_strip)] but not used `text.strip_prefix("0x")`.